torch/csrc/jit/codegen/cuda/codegen.cpp

```diff
@@ -275,6 +276,52 @@ class CudaKernelGenerator : private kir::IrVisitor {
   }
 
   void visit(const kir::UnaryOp* node) final {
+    bool is_vector_op = false;
+    size_t vector_word_size = 1;
+
+    if (node->out()->isA<kir::TensorIndex>()) {
+      auto ti = node->out()->as<kir::TensorIndex>();
+      for (auto id : ti->view()->fuserTv()->domain()->domain()) {
```

Comment:
Could we just if check if the last IterDomain is ParallelType::Vectorize?

Reply:
We should actually guarantee that it's the inner most split of the inner most root dim. We can't simply check that it's the last ID in the current domain view as the axes could have been reordered.

Reply:
Sure, but is it necessary to validate this check in the codegen?

Also, this loop didn't check that for condition?

Reply:
Sorry, I thought this question was in the validation not the codegen. Which we do need to implement that check there.

Reply:
What I was making here was a simpler "is this expression vectorized" in the codegen.

Reply:
I just wanted to keep the logic of figuring out how to generate the code, closer to generating the code.

Reply:
I'm confused. If we enforce the invariant "only allow ParallelType::Vectorize on the inner-most split of the inner-most merged root dim", couldn't we just check the last iterDomain?